pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/RealtimeKinesisIntegrationTest.java

```diff
@@ -208,15 +221,11 @@ public void startKinesis()
       throws Exception {
     try {
       final LocalstackDockerConfiguration dockerConfig = PROCESSOR.process(this.getClass());
-      localstackDocker.startup(dockerConfig);
-    } catch (LocalstackDockerException e) {
       StopAllLocalstackDockerCommand stopAllLocalstackDockerCommand = new StopAllLocalstackDockerCommand();
       stopAllLocalstackDockerCommand.execute();
-
-      final LocalstackDockerConfiguration dockerConfig = PROCESSOR.process(this.getClass());
-      //Restart localstack docker after killing all the existing localstack docker containers.
-      //If it fails at this step, the test is terminated.
       localstackDocker.startup(dockerConfig);
+    } catch (LocalstackDockerException e) {
```

Comment:
just skip the catch here and throw the exception up the stack?

Reply:
Yes, incorporated this change now. Earlier this catch block had logic to re-run the docker container in case it failed to start. But that is not required now with the new logic.